### PR TITLE
✨ | Adding cooldowns in Slash Commands (Interaction)

### PR DESCRIPTION
**Pull request by TecEash1**
Added a cooldown config for slash commands

~~- Buttons can now have multiple folders/subfolders~~

~~- More events can be added easily, e.g. channelCreate~~

### CODE REVIEW

**Comment by TecEash1** [comment]
I am not sure if this still allows for typescript assistant support... but I thought I would make a pr to help out. 
sorry if it doesn't 😅

File: bot.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const {
 	Collection,
 	GatewayIntentBits,
 	Partials,
+	ActivityType,
```
Comment (StormPacer) [inline bot.js:16]: ActivityType is not used in the code, it should be removed.

File: bot.js [hunk 2]
```diff
@@ -28,9 +29,9 @@ const client = new Client({
 	// Please add all intents you need, more detailed information @ https://ziad87.net/intents/
 	intents: [
 		GatewayIntentBits.Guilds,
-		GatewayIntentBits.DirectMessages,
 		GatewayIntentBits.GuildMessages,
 		GatewayIntentBits.MessageContent,
+		GatewayIntentBits.DirectMessages,
```
Comment (StormPacer) [inline bot.js:34]: Not sure why this was switched around, not that it matters but still a weird change.

File: bot.js [hunk 3]
```diff
@@ -41,23 +42,20 @@ const client = new Client({
...
 for (const file of eventFiles) {
 	const event = require(`./events/${file}`);
 	if (event.once) {
 		client.once(event.name, (...args) => event.execute(...args, client));
 	} else {
-		client.on(
-			event.name,
-			async (...args) => await event.execute(...args, client)
-		);
+		client.on(event.name, async (...args) => await event.execute(...args, client));
```
Comment (StormPacer) [inline bot.js:56]: Nothing was changed about the events part of the code, you should follow the repository's formatting using prettier.

File: bot.js [hunk 4]
```diff
@@ -167,24 +165,28 @@ for (const folder of contextMenus) {
...
+  for (const file of files) {
+    const path = `${dir}/${file}`;
+    if (fs.statSync(path).isDirectory()) {
+      walk(path);
+    } else if (file.endsWith('.js')) {
+      const command = require(path);
+      buttonCommands.push(command);
+    }
+  }
+}
```
Comment (StormPacer) [inline bot.js:182]: This seems redundant, I see no use for folders within folders in the button handler.

**Review by StormPacer — CHANGES_REQUESTED** [review]
The cooldown code works, the @version number in `slashCreate.js` needs to be changed to the new version number of the bot (I recommend the new version to be 3.4.0). You also need to add an example to the `help.js` slash command so people know how you can use cooldowns.

However, there are a lot of random commits that don't quite fit with the template or are not good to have in the template. I recommend looking over my comments and fixing accordingly.

File: bot.js (hunk 4 above)
Comment (TecEash1) [reply bot.js:182]: 👍, I just use it quite a bit as I have quite a lot of buttons. 
- I’ll remove that tho.

File: bot.js (hunk 2 above)
Comment (NamVr) [reply bot.js:34]: I don't understand the change, was DM intents not a part of code earlier so it is added now?
Also are we using DM commands by the way so it needs to be edited?

File: bot.js (hunk 3 above)
Comment (NamVr) [reply bot.js:56]: Why are you adding author tags everywhere editing nothing 🤔

**Comment by NamVr** [comment]
I've personally checked the commits and I think bot.js has gone with no valid changes and it should be returned to the default state before the PR (with no changes),
For slash.js command cooldown can be a good feature, however the code needs to be optimised to reduce bloat.
Lastly please use prettier as default formatter.

**Comment by TecEash1** [comment]
> I've personally checked the commits and I think bot.js has gone with no valid changes and it should be returned to the default state before the PR (with no changes), For slash.js command cooldown can be a good feature, however the code needs to be optimised to reduce bloat. Lastly please use prettier as default formatter.

👍Ill remove everything else and make the cooldown better, like the other one in the prefix commands

File: bot.js (hunk 2 above)
Comment (TecEash1) [reply bot.js:34]: o sorry about that, I was using it for something else and removed it and added it back... ill fix it

**Review by NamVr — CHANGES_REQUESTED** [review]
Please use prettier properly, like said below by @StormPacer

File: events/slashCreate.js [hunk 5]
```diff
@@ -1,43 +1,68 @@
...
 
-		// If the interaction is not a command in cache.
+    // Cooldowns
+    const { cooldowns } = client;
+    if (!cooldowns.has(command.name)) {
+      cooldowns.set(command.name, new Collection());
+    }
+    const now = Date.now();
+    const timestamps = cooldowns.get(command.name);
+    const defaultCooldownDuration = 0;
```
Comment (NamVr) [inline events/slashCreate.js:37]: Why do we need this, zero can directly be written in next line?

**Comment by NamVr** [comment]
Why did you force pushed branch? It wasnt' necessary?

**Comment by TecEash1** [comment]
> Why did you force pushed branch? It wasnt' necessary?

to keep it all in 1 commit

**Comment by TecEash1** [comment]
> Please use prettier properly, like said below by @StormPacer

I put it all into prettier and that's how it formatted it...

**Comment by NamVr** [comment]
It wasn't necessary since I always squash and merge.
I'm pretty sure it's the wrong formatting for prettier since all files are changed.
Do you mind if I close the PR and update this new command myself?

**Comment by TecEash1** [comment]
> It wasn't necessary since I always squash and merge.
> I'm pretty sure it's the wrong formatting for prettier since all files are changed.
> Do you mind if I close the PR and update this new command myself?

O ok, I used the prettier website and it seemed to format it like that for me. 
I think u have perms to make commits to the fork to update it tho

**Comment by TecEash1** [comment]
> Prettier website? I don't know what you're talking about, but there exists a Prettier extension on VS Code which is officially supported by us.
> Go to VS Code settings after installation of the above extension and then make it as default formatter for your code, make sure your code auto formats after every save.
> Then format your files by saving them it should solved all issues.

k ill try that, it said on [their website](https://prettier.io/) that it was the same, sorry about that

**Comment by TecEash1** [comment]
image
image
image

**Comment by NamVr** [comment]
I think most of it is now prettified and fix, does the cooldown property support Intellisense autocompletion? 
You might have to add property in typings.d.ts of "cooldown" in slash commands.

Otherwise, requesting review from @StormPacer !

**Comment by NamVr** [comment]
Hi, you were requested to complete the PR few days ago, please do it soon or we will have to close it.
(Can't review unless the full feature is completed)

**Comment by TecEash1** [comment]
> Hi, you were requested to complete the PR few days ago, please do it soon or we will have to close it.
> (Can't review unless the full feature is completed)

O ok, I thought storm was gonna do it that was y, mb 😅. I’ll try figure out how to add it tmr. Havnt rly looked into ts

**Comment by TecEash1** [comment]
image
Would that be how you would add it...?

**Comment by NamVr** [comment]
Yes it was really that simple.
Just check the cooldown property in legacy command on how it is written.

**Comment by TecEash1** [comment]
would you like me to change anything else?

**Comment by NamVr** [comment]
No, you're really trying to waste our times here I believe hence I have closed this request. This is a PR not a guide or a tutorial, you're expected to open a PR if you want to contribute, if we have to help you to be able to you to contribute the PR makes no sense and I can create the feature myself too.

**Comment by TecEash1** [comment]
k, I'm sorry for waisting your time

**Pull request closed without merging**